src/conn_handler.c

```diff
@@ -288,22 +288,42 @@ static void* flush_thread(void *arg) {
     // Cast the args
     metrics *m = arg;
 
+    struct timeval tv_start, tv_end, tv_diff;
+
     // Get the current time
-    struct timeval tv;
-    gettimeofday(&tv, NULL);
+    gettimeofday(&tv_start, NULL);
 
     // Determine which callback to use
     stream_callback cb = (GLOBAL_CONFIG->binary_stream)? stream_formatter_bin: stream_formatter;
 
     // Stream the records
-    int res = stream_to_command(m, &tv, cb, GLOBAL_CONFIG->stream_cmd);
+    int res = stream_to_command(m, &tv_start, cb, GLOBAL_CONFIG->stream_cmd);
     if (res != 0) {
         syslog(LOG_WARNING, "Streaming command exited with status %d", res);
     }
 
     // Cleanup
     destroy_metrics(m);
     free(m);
+
+    // Get time passed so that we can report on
+    // how long it took us to flush the metrics
+    gettimeofday(&tv_end, NULL);
+    int tv_diff_usec = ((tv_end.tv_sec * 1000000) + tv_end.tv_usec) - ((tv_start.tv_sec * 1000000) + tv_start.tv_usec);
```

Comment:
Suggestion (to avoid overflows maybe):
```
(tv_end.tv_sec - tv_start.tv_sec) * 1000000 + (tv_end.tv_usec - tv_start.tv_usec)
```

Reply:
Come to think of it... If our preferred unit is milliseconds, perhaps we should use the `double` data type and then do:
```
(tv_end.tv_sec - tv_start.tv_sec) * 1000 + (tv_end.tv_usec - tv_start.tv_usec) / 1000
```

Reply:
(or maybe truncate the microseconds and just keep the millis)